backend/Testing/FwHeadless/MediaFileServiceTests.cs

```diff
@@ -143,6 +149,101 @@ public async Task Sync_PreExistingFilesArePreserved()
         await AssertDbFileExists("SomeFile.txt");
     }
 
+    [Fact]
+    public async Task SaveMediaFile_Works()
+    {
+        var mediaFile = new MediaFile
+        {
+            Filename = "test.txt",
+            ProjectId = _projectId
+        };
+        await _service.SaveMediaFile(mediaFile, SimpleStream("test"));
+
+        await AssertDbFileExists("test.txt");
+        var filePath = _service.FilePath(mediaFile);
+        Directory.GetFiles(Path.GetDirectoryName(filePath)!).Should().Contain(filePath);
+        (await File.ReadAllTextAsync(filePath)).Should().Be("test");
+    }
+
+    private Stream SimpleStream(string content)
+    {
+        var memoryStream = new MemoryStream();
+        using (var stream = new StreamWriter(memoryStream, leaveOpen: true))
+        {
+            stream.Write(content);
+        }
+
+        memoryStream.Position = 0;
+        return memoryStream;
+    }
+
+    [Fact]
+    public async Task SaveMediaFile_HgRollbackDoesNotDeleteFile()
+    {
+        var mediaFile = new MediaFile
+        {
+            Filename = "test.txt",
+            ProjectId = _projectId
+        };
+        await _service.SaveMediaFile(mediaFile, SimpleStream("test"));
+
+        var filePath = _service.FilePath(mediaFile);
+        var directoryName = Path.GetDirectoryName(filePath)!;
+        Directory.GetFiles(directoryName).Should().Contain(filePath);
+
+        //make a commit which would be rolled back by S&R during a conflict
+        var conflictFile = Path.Join(_cache.LangProject.LinkedFilesRootDir, "conflict.txt");
+        await File.WriteAllTextAsync(conflictFile, "test");
+        HgRunner.Run("hg commit --message \"test commit\" --addremove", directoryName, 5, new NullProgress());
+
+        var rev = int.Parse(_hgRepository.GetHeads().Single().Number.LocalRevisionNumber) - 1;
+        //simulate rollback as seen here: https://github.com/sillsdev/chorus/blob/af6e5c0e97758aef00bd2104b6c1ccf5719798ef/src/LibChorus/sync/Synchronizer.cs#L574
+        _hgRepository.RollbackWorkingDirectoryToRevision(rev.ToString());
```

Comment:
@jasonleenaylor I'm just guessing here, this doesn't match exactly because I'm taking the head and subtracting one in order to rollback what I'm assuming would be the commit which happens locally before any changes are pulled? But I'm not sure what the difference is between this and whats happening during a conflict.